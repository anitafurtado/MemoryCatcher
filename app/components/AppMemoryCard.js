import React from 'react';
import { Image, StyleSheet, TouchableOpacity, View } from 'react-native';

import AppColors from '../config/AppColors';
import AppText from './AppText';


function AppMemoryCard({title, category, date, location, image, onPress}) {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.container}>
        <View style={styles.topText}>
          <AppText>{date}</AppText>
          <AppText>{category}</AppText>
        </View>
        {isFinite(image) ? <Image source={image} style={styles.image}/> : <Image source={{uri: image}} style={styles.image}/>}
        <View style={styles.bottomText}>
          <AppText>{title}</AppText>
          <AppText>{location}</AppText>
        </View>
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container:{
    backgroundColor: AppColors.secondaryVariantColor,
    margin: 10,
    borderRadius: 5,
    borderWidth: 1,
  },
  topText:{
    flexDirection: "row",
    justifyContent: "space-between",
    padding: 5,
  },
  image:{
    width: 250,
    height: 150,
    alignSelf: "center",
  },
  bottomText:{
    flexDirection: "row",
    justifyContent: "space-between",
    padding: 5,
  },
})

export default AppMemoryCard;