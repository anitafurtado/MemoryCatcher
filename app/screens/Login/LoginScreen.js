import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Formik } from 'formik';
import * as Yup from 'yup';

import AppButton from '../../components/AppButton';
import AppColors from '../../config/AppColors';
import AppScreen from '../../components/AppScreen';
import AppText from '../../components/AppText';
import AppTextInput from '../../components/AppTextInput';
import AppTextInputError from '../../components/AppTextInputError';
import DataManager from '../../config/DataManager';

// Validation Schema
const schema = Yup.object().shape(
  {
    email: Yup.string().required().email().label("Email"),
    password: Yup.string().required().min(4).max(10).label("Password"),
  }
);

//Check if User information is correct
const validateUser = ({email, password}) => {
  let data = DataManager.getInstance();
  return data.validateUser(email, password);
};


function LoginScreen({navigation, route}) {
  const message = () =>{
    return route.params.paramMessage !==undefined;
  }
  return (
    <AppScreen>
      {message ? <AppText style={{textAlign: "center"}}>{route.params.paramMessage}</AppText> : <></>}
      <View style = {styles.container}>
        
        <AppText>Please Enter Your Login Details:</AppText>
        <Formik
          initialValues={{email: '', password: ''}}
          onSubmit={(values)=>{
            if(validateUser(values)) {
              let data = DataManager.getInstance();
              let user = data.getUser(values.email);
              data.setUserID(user.userID);
                navigation.navigate("HomeWithTab");
                            
            } else {
              alert("Invalid email or password. Please try again.")
            }
          }
            
          }
          validationSchema={schema}
          
        >
          {({handleChange, handleSubmit, errors, setFieldTouched, touched})=>(
            <>
              <View style = {styles.form}>
                <AppTextInput
                  placeholder="Email"
                  onChangeText = {handleChange("email")}
                  autoCapitalize="none"
                  autoCorrect={false}
                  keyboardType="email-address"
                  onBlur = {() => setFieldTouched("email")}
                />
                {touched.email && <AppTextInputError>{errors.email}</AppTextInputError>}
                <AppTextInput
                  placeholder="Password"
                  onChangeText = {handleChange("password")}
                  autoCapitalize="none"
                  autoCorrect={false}
                  secureTextEntry
                  onBlur = {() => setFieldTouched("password")}
                />
                {touched.password && <AppTextInputError>{errors.password}</AppTextInputError>}
                <AppButton title="Login" onPress ={handleSubmit}/>
              </View>
              
            </>
          )}
        </Formik>

      </View>
      
      
    </AppScreen>
  );
}

const styles = StyleSheet.create({ 
  container:{
    alignItems: "center",
    backgroundColor: AppColors.primaryColor,
    flex: 1,
    justifyContent: "center",
    
  },
  form:{
    alignItems: "center",
    justifyContent: "center",
  }
  
})

export default LoginScreen;