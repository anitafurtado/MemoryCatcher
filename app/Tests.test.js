import React from "react";
import renderer from "react-test-renderer";
import LoginScreen from "./screens/Login/LoginScreen";
import AppMemoryCard from "./components/AppMemoryCard";
import { render, fireEvent } from '@testing-library/react-native';

//1. Style Test

test("Correct AppMemoryCard style", () => {
  const json = renderer.create(
    <AppMemoryCard
      title="Memory Title"
      category="Memory Category"
      date="Memory date"
      location="Memory location"
      image="Memory image"
      onPress={() => {
        console.log("onPress Function");
      }}
    />
  ).toJSON();

  //container
  expect(json.children[0].props.style.backgroundColor).toBe("#C397BB");
  expect(json.children[0].props.style.margin).toBe(10);
  expect(json.children[0].props.style.borderRadius).toBe(5);
  expect(json.children[0].props.style.borderWidth).toBe(1);

  //Top Text
  expect(json.children[0].children[0].props.style.flexDirection).toBe("row");
  expect(json.children[0].children[0].props.style.justifyContent).toBe("space-between");
  expect(json.children[0].children[0].props.style.padding).toBe(5);

  // Image
  expect(json.children[0].children[1].props.style.width).toBe(250);
  expect(json.children[0].children[1].props.style.height).toBe(150);
  expect(json.children[0].children[1].props.style.alignSelf).toBe("center");
  
  //Bottom Text
  expect(json.children[0].children[2].props.style.flexDirection).toBe("row");
  expect(json.children[0].children[2].props.style.justifyContent).toBe("space-between");
  expect(json.children[0].children[2].props.style.padding).toBe(5);

  //App Text for Date
  expect(json.children[0].children[0].children[0].props.style[0].fontFamily).toBe("Arial");
  expect(json.children[0].children[0].children[0].props.style[0].fontSize).toBe(18);

  //App Text for Category
  expect(json.children[0].children[0].children[1].props.style[0].fontFamily).toBe("Arial");
  expect(json.children[0].children[0].children[1].props.style[0].fontSize).toBe(18);

  //App Text for Title
  expect(json.children[0].children[2].children[0].props.style[0].fontFamily).toBe("Arial");
  expect(json.children[0].children[2].children[0].props.style[0].fontSize).toBe(18);

  //App Text for Location
  expect(json.children[0].children[2].children[1].props.style[0].fontFamily).toBe("Arial");
  expect(json.children[0].children[2].children[1].props.style[0].fontSize).toBe(18);
});

//2. Snapshot Tests

test("AppMemoryCard with no props renders correctly", () => {
  const tree = renderer.create(<AppMemoryCard />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("AppMemoryCard with all props but no onPress function renders correctly", () => {
  const tree = renderer.create(
    <AppMemoryCard
      title="Memory Title"
      category="Memory Category"
      date="Memory date"
      location="Memory location"
      image="Memory image"
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test("AppMemoryCard with all props renders correctly", () => {
  const tree = renderer.create(
    <AppMemoryCard
      title="Memory Title"
      category="Memory Category"
      date="Memory date"
      location="Memory location"
      image="Memory image"
      onPress={() => {
        console.log("onPress Function");
      }}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Login Screen renders correctly", () => {
  const navigation = { navigate: jest.fn() };
  const route = { params: jest.fn() };
  const tree = renderer.create(
    <LoginScreen 
      navigation={navigation} 
      route={route} 
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

//3. Other Login Screen Components

test("Empty Register Message when user has not Registered first", () => {
  const navigation = { navigate: jest.fn() };
  const route = { params: jest.fn() };
  const json = renderer.create(
    <LoginScreen 
      navigation={navigation} 
      route={route} 
    />
  ).toJSON();
  expect(json.children[0].children).toBe(null);
});

test("Header text for Login Screen", () => {
  const navigation = { navigate: jest.fn() };
  const route = { params: jest.fn() };
  const json = renderer.create(
    <LoginScreen 
      navigation={navigation} 
      route={route} 
    />
  ).toJSON();
  expect(json.children[1].children[0].children.includes("Please Enter Your Login Details:"));
  expect(json.children[1].children[0].props.style[0].fontFamily).toBe("Arial");
  expect(json.children[1].children[0].props.style[0].fontSize).toBe(18);
});

test("Formik Text Input for Email", () => {
  const navigation = { navigate: jest.fn() };
  const route = { params: jest.fn() };
  const json = renderer.create(
    <LoginScreen 
      navigation={navigation} 
      route={route} 
    />
  ).toJSON();
  expect(json.children[1].children[1].children[0].props.placeholder).toBe("Email");
  expect(json.children[1].children[1].children[0].props.autoCapitalize).toBe("none");
  expect(json.children[1].children[1].children[0].props.autoCorrect).toBe(false);
  expect(json.children[1].children[1].children[0].props.keyboardType).toBe("email-address");

});

test("Formik Text Input for Password", () => {
  const navigation = { navigate: jest.fn() };
  const route = { params: jest.fn() };
  const json = renderer.create(
    <LoginScreen 
      navigation={navigation} 
      route={route} 
    />
  ).toJSON();
  expect(json.children[1].children[1].children[1].props.placeholder).toBe("Password");
  expect(json.children[1].children[1].children[1].props.autoCapitalize).toBe("none");
  expect(json.children[1].children[1].children[1].props.autoCorrect).toBe(false);
  expect(json.children[1].children[1].children[1].props.secureTextEntry).toBe(true);

});

test("Login AppButton", () => {
  const navigation = { navigate: jest.fn() };
  const route = { params: jest.fn() };
  const json = renderer.create(
    <LoginScreen 
      navigation={navigation} 
      route={route} 
    />
  ).toJSON();
  expect(json.children[1].children[1].children[2].children[0].children[0].children.includes("Login"));  

});