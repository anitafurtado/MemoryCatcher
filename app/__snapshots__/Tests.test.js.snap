// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMemoryCard with all props but no onPress function renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#C397BB",
        "borderRadius": 5,
        "borderWidth": 1,
        "margin": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Arial",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        Memory date
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Arial",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        Memory Category
      </Text>
    </View>
    <Image
      source={
        Object {
          "uri": "Memory image",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 150,
          "width": 250,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Arial",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        Memory Title
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Arial",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        Memory location
      </Text>
    </View>
  </View>
</View>
`;

exports[`AppMemoryCard with all props renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#C397BB",
        "borderRadius": 5,
        "borderWidth": 1,
        "margin": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Arial",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        Memory date
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Arial",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        Memory Category
      </Text>
    </View>
    <Image
      source={
        Object {
          "uri": "Memory image",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 150,
          "width": 250,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Arial",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        Memory Title
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Arial",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        Memory location
      </Text>
    </View>
  </View>
</View>
`;

exports[`AppMemoryCard with no props renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#C397BB",
        "borderRadius": 5,
        "borderWidth": 1,
        "margin": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Arial",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Arial",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      />
    </View>
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 150,
          "width": 250,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Arial",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Arial",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Login Screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#5985B1",
        "flex": 1,
        "marginTop": 0,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Arial",
          "fontSize": 18,
        },
        Object {
          "textAlign": "center",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5985B1",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Arial",
            "fontSize": 18,
          },
          undefined,
        ]
      }
    >
      Please Enter Your Login Details:
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        keyboardType="email-address"
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="#000"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "borderRadius": 20,
              "borderWidth": 1,
              "margin": 10,
              "padding": 10,
              "width": 200,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#000"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "borderRadius": 20,
              "borderWidth": 1,
              "margin": 10,
              "padding": 10,
              "width": 200,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#B279A7",
                "borderRadius": 10,
                "borderWidth": 2,
                "justifyContent": "center",
                "margin": 10,
                "padding": 10,
                "width": 100,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Arial",
                  "fontSize": 18,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                  undefined,
                ],
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
